src/ar-data-bundle.ts

```diff
@@ -66,7 +66,8 @@ export async function bundleAndSignData(dataItems: DataItem[], signer: Signer):
 export async function getSignatureAndId(item: DataItem, signer: Signer): Promise<{ signature: Buffer; id: Buffer }> {
   const signatureData = await getSignatureData(item);
 
-  const signatureBytes = await signer.sign(signatureData);
+  // some signers, like arconnect and arweave.app, require the entire item to be passed in
+  const signatureBytes = await signer.sign(signatureData, { dataItem: item });
```

Comment:
We need to pass in the actual item here. The item is passed in as dataItem or transaction for ArconnectSigner to be able to decide which method to use to generate the signature.